luskan/src/main/java/org/opencadc/luskan/ws/StorageTapService.java

```diff
@@ -148,38 +155,86 @@ public AvailabilityStatus getStatus() {
 
-            URI groupsURI = localAuthority.getServiceURI(Standards.GMS_SEARCH_01.toString());
-            url = reg.getServiceURL(groupsURI, Standards.VOSI_AVAILABILITY, AuthMethod.ANON).toExternalForm();
+            url = getAvailabilityForLocal(Standards.GMS_SEARCH_01);
             checkResource = new CheckWebService(url);
             checkResource.check();
 
         } catch (CheckException ce) {
             // tests determined that the resource is not working
             isGood = false;
             note = ce.getMessage();
-        } catch (Throwable t) {
+        } catch (Throwable throwable) {
             // the test itself failed
-            log.error("test failed", t);
+            StringBuilder sb = new StringBuilder();
+            sb.append("test failed because: ");
+            while (throwable != null) {
+                sb.append(throwable.getMessage()).append(": ");
+                throwable = throwable.getCause();
+            }
+            log.error(sb.toString());
             isGood = false;
-            note = "test failed, reason: " + t;
+            note = sb.toString();
         }
         return new AvailabilityStatus(isGood, null, null, null, note);
     }
 
+    private String getAvailabilityForLocal(URI standardID)
+        throws ResourceNotFoundException {
+        LocalAuthority localAuthority = new LocalAuthority();
+        RegistryClient reg = new RegistryClient();
+
+        URI resourceID;
+        try {
+            resourceID = localAuthority.getServiceURI(standardID.toString());
+        } catch (NoSuchElementException e) {
+            String message = String.format("service URI not found in LocalAuthority for standardID %s", standardID);
+            throw new ResourceNotFoundException(message);
+        }
+
+        Capabilities capabilities;
+        try {
+            capabilities = reg.getCapabilities(resourceID);
+        } catch (IOException e) {
+            String message = String.format("io error getting capabilities for resourceID %s, standardID %s, because %s",
+                                           resourceID, standardID, e.getMessage());
+            throw new ResourceNotFoundException(message);
+        }
+
+        Capability capability = capabilities.findCapability(Standards.VOSI_AVAILABILITY);
+        if (capability == null) {
+            String message =
+                String.format("service %s not does not provide %s", standardID, Standards.VOSI_AVAILABILITY);
+            throw new UnsupportedOperationException(message);
+        }
+
+        Interface anInterface = capability.findInterface(AuthMethod.ANON);
+        if (anInterface == null) {
+            String message = String.format(
+                "unexpected: service %s capability %s does not support auth: %s",
+                standardID, capability, AuthMethod.ANON);
+            throw new RuntimeException(message);
+        }
+
+        AccessURL accessURL = anInterface.getAccessURL();
```

Comment:
The accessURL cannot be null so just return it here.

Reply:
Updated.